### PR TITLE
Support the Ninja Generator, address warnings for newer CMake

### PR DESCRIPTION
**Pull request by thewtex**

### CODE REVIEW

**Comment by thewtex** [comment]
@jcfr @jessdtate please take a look.

File: src/test/CMakeLists.txt [hunk 1]
```diff
@@ -12,13 +12,21 @@ include(ExternalProject)
 
 set(googletest_GIT_TAG "origin/cleaver")
 
+SET(_build_byproducts)
+IF(NOT WIN32)
+  FOREACH(_lib gtest gtest_main gmock gmock_main)
+    LIST(APPEND _build_byproducts ${GTEST_INSTALL_DIR}/lib/lib/${CMAKE_FIND_LIBRARY_PREFIXES}${_lib}${CMAKE_STATIC_LIBRARY_SUFFIX})
+  ENDFOREACH()
+ENDIF()
```
Comment (jcfr) [inline src/test/CMakeLists.txt:20]:
```suggestion
set(_build_byproducts)
if(NOT WIN32)
  foreach(_lib IN ITEMS gtest gtest_main gmock gmock_main)
    list(APPEND _build_byproducts ${GTEST_INSTALL_DIR}/lib/lib/${CMAKE_FIND_LIBRARY_PREFIXES}${_lib}${CMAKE_STATIC_LIBRARY_SUFFIX})
  endforeach()
endif()
```

**Comment by allywarner** [comment]
@thewtex This fixes the Ninja build issue with gtest. However, the Travis builds were building before and should pass before merging.

**Comment by thewtex** [comment]
@allywarner @jcfr thanks for the updates and reviews.

I pushed an update to Travis to go from Ubuntu 14.04 to 18.04. Let's see how it goes.

**Comment by thewtex** [comment]
Back to green again :green_apple: :rocket: . This should be ready now.

**Review by jessdtate — DISMISSED** [review]

**Review by allywarner — APPROVED** [review]

**Comment by jcfr** [comment]
While integrating, I suggest to use "Squash and merge" and setup the following commit title:

```
COMP: Add support for Ninja CMake Generator (#104)

* Update googletest external project specifying BUILD_BYPRODUCTS required by Ninja generator.
  It addressed this error:

     ninja: error: 'externals/googletest/lib/lib/libgmock_main.a', needed by 'test/bin/linearviolationchecker_tests', missing and no known rule to make it

* Updates the minimum required version of CMake from 2.6 to 3.10.2. CMake 3.10.2 is
  the version available with Ubuntu 18.04, long term release. This also the current minimum
  required version for ITK.

* Update TravisCI image to ensure a more recent version of CMake is available.

* For future reference, this commit addresses the following CMake warnings:

  CMake Warning (dev) at /home/matt/src/CMake/Modules/CTest.cmake:50 (option):
    Policy CMP0077 is not set: option() honors normal variables.  Run "cmake
    --help-policy CMP0077" for policy details.  Use the cmake_policy command to
    set the policy and suppress this warning.

    For compatibility with older versions of CMake, option is clearing the
    normal variable 'BUILD_TESTING'.
  Call Stack (most recent call first):
    Cleaver2/src/CMakeLists.txt:104 (INCLUDE)
  This warning is for project developers.  Use -Wno-dev to suppress it.


  CMake Deprecation Warning at lib/jsoncpp/CMakeLists.txt:3 (CMAKE_MINIMUM_REQUIRED):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.
  
    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.
  
  CMake Deprecation Warning at lib/cleaver/CMakeLists.txt:35 (CMAKE_MINIMUM_REQUIRED):
    Compatibility with CMake < 2.8.12 will be removed from a future version of
    CMake.
  
    Update the VERSION argument <min> value or use a ...<max> suffix to tell
    CMake that the project does not need compatibility with older versions.

Co-authored-by: Matt McCormick <matt.mccormick@kitware.com>
Co-authored-by: Jean-Christophe Fillion-Robin <jcfr@kitware.com>
```

**Comment by jcfr** [comment]
Thanks @allywarner  :pray:

**Pull request merged**